superset/datasets/commands/delete.py

```diff
@@ -43,19 +41,13 @@ def __init__(self, model_id: int):
 
     def run(self) -> Model:
         self.validate()
-        self._model = cast(SqlaTable, self._model)
+        assert self._model
+
         try:
-            # Even though SQLAlchemy should in theory delete rows from the association
```

Comment:
@betodealmeida this should address the problem you were trying to circumvent in https://github.com/apache/superset/pull/23414.

Fun fact I wondered if I wrote this comment as it's somewhat in the style I write, but I realized I don't typically use the term "let's".